### ISSUE TITLE
How do you set coverage threshold?

### ISSUE THREAD

**Issue by melxx001** [monounity/karma-typescript#80]
I'm trying to set a code coverage threshold. In karma-coverage, you can do so by adding the following to karma.conf.js (https://github.com/karma-runner/karma-coverage/blob/master/docs/configuration.md)

```
coverageReporter: {
  check: {
    global: {
      statements: 50,
      branches: 50,
      functions: 50,
      lines: 50,
      excludes: [
        'foo/bar/**/*.js'
      ]
    },
    each: {
      statements: 50,
      branches: 50,
      functions: 50,
      lines: 50,
      excludes: [
        'other/directory/**/*.js'
      ],
      overrides: {
        'baz/component/**/*.js': {
          statements: 98
        }
      }
    }
  }
}
```

it works when I add `reporters: ['spec', 'karma-typescript', 'coverage'],` but the coverage seems to be calculated on the transpiled JS and not the typescript files (which gives a wrong coverage summary)

**Comment by adamholdenyall** [comment]
For what it's worth, I was able to use the [karma-istanbul-threshold](https://www.npmjs.com/package/karma-istanbul-threshold) package. You have to generate a json report with karma-typescript, then feed the JSON file into this reporter and it will check thresholds, set the proper error level etc. I will say that I think the README example is currently wrong and that "istanbul-threshold" should be added to the reporters, not "karma-istanbul-threshold".

**Comment by melxx001** [comment]
@adamholdenyall Thanks :) I may end up doing that but I wouldn't need karma-typescript anymore If I'm going the use that, with karma-remap-istanbul etc... I enjoyed that I don't have map or remap anything

@erikbarke I just noticed you put an enhancement tag. Would you be able to give me an ETA on when that could be available?

**Comment by erikbarke** [comment]
@adamholdenyall, thanks!

@melxx001, I'm expecting to release within a month!

**Comment by erikbarke** [comment]
Hey guys, I've added thresholds for remapped coverage:

```js
karmaTypescriptConfig: {
    coverageOptions: {
        threshhold: {
            global: {
                statements: 100,
                branches: 100,
                functions: -10,
                lines: 100,
                excludes: [
                    "src/foo/**/*.js"
                ]
            },
            file: {
                statements: -10,
                branches: 100,
                functions: 100,
                lines: 100,
                excludes: [
                    "src/bar/**/*.js"
                ],
                overrides: {
                    "src/file.js": {
                        statements: 90
                    }
                }
            }
        },
    }
}
```

A positive value means min percentage of coverage, a negative value means max number of uncovered items, excludes and overrides are globbing patterns.

It's on the the latest dev branch now, `npm install https://github.com/monounity/karma-typescript.git#2.1.8` 😃

**Comment by wawyed** [comment]
@erikbarke That's just brilliant! Looking forward to the release, I'm wondering how this would interact with  https://github.com/monounity/karma-typescript/issues/97 If I also wanted to the coverage threshold for javascript files.

**Comment by erikbarke** [comment]
@wawyed, please see my answer in #97!

**Comment by erikbarke** [comment]
3.0.0-beta.1 is on npm now with support for coverage thresholds:

`npm install karma-typescript@beta`

(2.1.8 ended up having so much new functionality + breaking changes it was renamed to 3.0.0)

**Comment by ehs035** [comment]
There is a little spelling error in your config sample:   **threshhold** should be **threshold**. I took me an hour to figure it out. Revenge of cut and paste coding.  :smiley:

**Comment by erikbarke** [comment]
@ehs035, fixed now, sorry 😞

**Comment by ehs035** [comment]
No worries, you gave me(us) a great tool. I use 3.0.0-beta.1 in production and it works like a charm.

**Comment by ShilpiAg1988** [comment]
I am unable to get coverage for TS files with these configurations. Can you please help?
